### ISSUE TITLE
details of parameter settings or training logs

### ISSUE THREAD

**Issue by Struggle-Forever** [4AI/AGN#6]
Hi, can you please provide us with details of parameter settings or training logs? The results we reproduced on multiple datasets are quite different from the results of the paper, and we need your help, thanks.  Our reproduced results of SST2 are as follows, AGN is not even as good as BERT-base.

![image](https://user-images.githubusercontent.com/29114869/151179161-71228064-dfa5-4dcd-8790-6d8423709d06.png)

**Comment by SeanLee97** [comment]
Sorry for the delaying reply.

I used the latest code to conduct the experiment on SST-2 dataset, the result is as follows:

```bash
iteration 1 accuracy: 0.93125, f1: 0.9311487929219376
iteration 2 accuracy: 0.92875, f1: 0.9287214255011527

Average accuracy:  0.930
Average f1: 0.9299351092115451
```

The results are slightly lower than the reported results because the prior experiments were trained under a fixed random seed. For general use, the random seed setting had been removed in the latest code.

The configure file is as follows:

```JSON
{
  "max_len": 64,
  "ae_epochs": 100,
  "epochs": 10,
  "batch_size": 32,
  "learning_rate": 0.00001,
  "pretrained_model_dir": "/mnt/pretrained-bert/uncased-bert-base/",
  "train_path": "datas//SST-2/train.jsonl",
  "dev_path": "datas/SST-2/test.jsonl",
  "save_dir": "sst2-model",
  "epsilon": 0.05,
  "iterations": 2,
  "verbose": 1
}
```

My environments:

```bash
$conda list | egrep 'tensorflow|keras'
keras                     2.3.1                         0
tensorflow                1.15.0          gpu_py37h0f0df58_0
tensorflow-base           1.15.0          gpu_py37h9dcbed7_0
tensorflow-estimator      1.15.1             pyh2649769_0
tensorflow-gpu            1.15.0               h0d30ee6_0
```

Tensorflow 1 and Tensorflow 2 might have different results. We recommend using TF1 because the code has not been tested in TensorFlow 2 yet.

**Comment by SeanLee97** [comment]
To achieve better results, you can try to tune the epsilon, learning rate, and dropout rate.

**Comment by SeanLee97** [comment]
Hi, I have added all of the tricks of my original implementation to the latest code. 

1) support to set random seed via `export RANDOM_SEED={NUM}`
2) AdamW optimizer
3) FGM attack

My configure file:

```JSON
{
  "max_len": 100,
  "ae_epochs": 100,
  "epochs": 50,
  "batch_size": 32,
  "learning_rate": 0.00003,
  "pretrained_model_dir": "/mnt/pretrained-bert/uncased-bert-base/",
  "train_path": "datas/SST-2/train.jsonl",
  "dev_path": "datas/SST-2/test.jsonl",
  "save_dir": "sst2-model",
  "dropout": 0.3,
  "apply_fgm": true,
  "fgm_rate": 0.3,
  "epsilon": 0.05,
  "iterations": 1,
  "verbose": 1
}
```

To run the latest code, you should set `TF_KERAS=1` first, e.g.,

```bash
export TF_KERAS=1; CUDA_VISIBLE_DEVICES=0 python main.py /path/to/config
```

Hope it helps!
